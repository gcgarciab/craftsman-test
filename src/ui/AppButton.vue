<template>
  <button
    class="text-white"
    :data-test="`${testName ? testName : 'button'}`"
    :class="[
      { 'mt-8': marginTop },
      [text ? 'px-10' : 'px-2'],
      [icon ? 'flex items-center justify-center' : 'inline'],
      color,
    ]"
    :disabled="disabled"
    :type="type"
    :title="focusText"
  >
    <fa
      v-if="icon"
      class="icon text-sm"
      :icon="icon"
      :class="{ 'mr-2': text }"
    />
    {{ text }}
  </button>
</template>

<script setup lang="ts">
import { PropType } from 'vue';
import ButtonType from '@/enums/button-type';
import ButtonColor from '@/enums/button-color';

const props = defineProps({
  type: {
    type: String as PropType<ButtonType>,
    default: ButtonType.BUTTON,
  },
  color: {
    type: String,
    default: ButtonColor.DEFAULT,
  },
  text: {
    type: String,
    required: false,
  },
  icon: {
    type: String,
    required: false,
  },
  marginTop: {
    type: Boolean,
    default: false,
  },
  disabled: {
    type: Boolean,
    default: false,
  },
  focusText: {
    type: String,
    required: false,
  },
  testName: {
    type: String,
    required: false,
  },
});
</script>

<style scoped>
button {
  @apply rounded-md py-2 mx-1 font-semibold shadow-sm transition duration-200;
}

button:focus {
  @apply ring ring-opacity-50;
}

button:disabled {
  @apply cursor-not-allowed opacity-60;
}

.default {
  @apply bg-gray-500 text-white border border-gray-500;
}

.default:focus {
  @apply ring-gray-300;
}

.default:hover {
  @apply bg-gray-600;
}

.simple {
  @apply p-0 bg-transparent border-none text-gray-600;
}

.primary {
  @apply bg-blue-500 border border-blue-500;
}

.primary:focus {
  @apply ring-blue-300;
}

.primary:hover {
  @apply bg-blue-600;
}

.success {
  @apply bg-green-500 border border-green-500;
}

.success:focus {
  @apply ring-green-300;
}

.success:hover {
  @apply bg-green-600;
}

.danger {
  @apply bg-red-500 border border-red-500;
}

.danger:focus {
  @apply ring-red-300;
}

.danger:hover {
  @apply bg-red-600;
}

button svg.icon {
  @apply w-4;
}
</style>
