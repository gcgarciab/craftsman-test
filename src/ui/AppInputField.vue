<template>
  <div class="input-block" :class="{ 'no-margin': noMargin }">
    <label v-if="label" class="block text-gray-700 mb-1">{{ label }}</label>

    <div v-if="type !== 'textarea'" class="input">
      <fa v-if="icon" class="input-icon" :icon="icon" />

      <input
        :type="inputType"
        :value="modelValue"
        :placeholder="placeholder"
        :class="{ 'with-icon': icon }"
        data-test="input-field"
        @input="updateValue($event)"
        @blur="$emit('checkErrors')"
      />

      <span v-if="extra" class="extra ml-2">{{ extra }}</span>

      <button
        v-if="type === 'password'"
        type="button"
        class="absolute right-3 top-2 text-gray-500 text-lg"
        @click="switchVisibility()"
      >
        <fa :icon="showPassword ? 'eye-slash' : 'eye'" />
      </button>
    </div>

    <div v-else class="input">
      <textarea
        rows="4"
        :value="modelValue"
        :placeholder="placeholder"
        @input="updateValue($event)"
        @blur="$emit('checkErrors')"
      ></textarea>
    </div>
  </div>
</template>

<script setup lang="ts">
import { ref } from 'vue';

const emit = defineEmits(['update:modelValue', 'checkErrors']);
const props = defineProps({
  modelValue: {
    type: String,
    required: true,
  },
  label: {
    type: String,
    required: false,
  },
  placeholder: {
    type: String,
    required: false,
  },
  icon: {
    type: String,
    required: false,
  },
  extra: {
    type: String,
    required: false,
  },
  type: {
    type: String,
    required: false,
    default: 'text',
  },
  noMargin: {
    type: Boolean,
    required: false,
  },
});

const inputType = ref(props.type);
const showPassword = ref(false);

/**
 * Receives input event and emit its value to parent.
 * @event => Input file event
 */
function updateValue(event: Event): void {
  const { value } = event.target as HTMLInputElement;
  const newValue = props.type === 'number' ? +value : value;

  emit('update:modelValue', newValue);
}

/**
 * Change input type to allow to see its value.
 */
function switchVisibility(): void {
  showPassword.value = !showPassword.value;
  inputType.value = showPassword.value ? 'text' : 'password';
}
</script>

<style scoped>
.input-block {
  @apply w-full text-left mb-6 mr-3;
}

.input-block:last-of-type {
  @apply mr-0;
}

textarea {
  @apply resize-none;
}

input,
textarea {
  @apply block w-full rounded-md border border-gray-300 shadow-sm px-4 py-2;
}

input:focus,
textarea:focus {
  @apply border-gray-300 ring ring-gray-200 ring-opacity-50;
}

.input {
  @apply relative flex items-center;
}

.input:last-of-type,
.input:last-of-type .extra {
  @apply mr-0;
}

.input-icon {
  @apply absolute left-3 top-3 text-gray-500 text-lg;
}

.with-icon {
  @apply pl-10;
}

.has-errors label {
  @apply text-red-500;
}

.has-errors input,
.has-errors textarea {
  @apply border-red-500;
}

.has-errors input:focus,
.has-errors textarea:focus {
  @apply ring-red-100;
}

.no-margin {
  @apply m-0;
}
</style>
