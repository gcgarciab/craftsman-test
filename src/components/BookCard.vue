<template>
  <div class="book-card">
    <div class="book-header">
      <div class="book-image w-36 h-48 flex justify-center items-center border">
        <img
          :src="volumeInfo.imageLinks?.smallThumbnail"
          alt="Book image"
          title="Book image"
        />
      </div>

      <div class="main-info">
        <h3 class="book-title">
          {{ volumeInfo.title }}

          <span v-if="volumeInfo.subtitle" class="book-subtitle">{{
            volumeInfo.subtitle
          }}</span>
        </h3>

        <h4 class="authors">
          <span
            v-for="(author, index) in volumeInfo.authors"
            :key="`author-${index + 1}`"
            >{{ author }}</span
          >
        </h4>

        <p v-if="volumeInfo.publisher" class="published mt-3">
          {{ volumeInfo.publisher }},
          {{ volumeInfo.publishedDate }}
        </p>

        <div v-if="volumeInfo.language" class="language mt-3">
          Language: ({{ volumeInfo.language.toUpperCase() }})

          <span v-if="volumeInfo.pageCount" class="total-pages mt-3">
            - Pages: {{ volumeInfo.pageCount }}
          </span>
        </div>
      </div>
    </div>

    <div
      class="card-actions absolute right-6 top-6 w-20 flex"
      :class="showDetailLink ? 'justify-between' : 'justify-end'"
    >
      <AppButton
        v-if="showDetailLink"
        icon="info-circle"
        focus-text="See more"
        :type="ButtonType.BUTTON"
        :color="ButtonColor.PRIMARY"
        @click="goToCurrentBook(id)"
      />

      <AppButton
        icon="heart"
        :focus-text="`${isFavorite ? 'Remove from' : 'Add to'} favorites`"
        :type="ButtonType.BUTTON"
        :color="isFavorite ? ButtonColor.DANGER : ButtonColor.DEFAULT"
        @click="emit('toggleFavorite')"
      />
    </div>
  </div>
</template>

<script setup lang="ts">
import { PropType } from 'vue';
import { useRouter } from 'vue-router';
import AppButton from '@/ui/AppButton.vue';
import ButtonType from '@/enums/button-type';
import ButtonColor from '@/enums/button-color';
import BookVolumeInfo from '@/interfaces/book/book-volume-info';

const router = useRouter();
const emit = defineEmits(['toggleFavorite']);
const props = defineProps({
  id: {
    type: String,
    required: true,
  },
  volumeInfo: {
    type: Object as PropType<Partial<BookVolumeInfo>>,
    required: true,
  },
  showDetailLink: {
    type: Boolean,
    required: false,
  },
  isFavorite: {
    type: Boolean,
    required: false,
  },
});

/**
 * Redirect to Book detail view.
 * @param id => Book id
 */
function goToCurrentBook(id: string): void {
  router.push({ name: 'BookDetail', params: { id } });
}
</script>

<style scoped>
.book-card {
  min-height: 16rem;
  @apply relative rounded-lg shadow-md px-6 py-10 w-full;
  @apply md:px-10 md:py-8 border my-6;
}

.book-header {
  @apply flex flex-col items-center;
  @apply md:flex-row md:items-start;
}

.book-title {
  @apply px-16 font-semibold;
  @apply md:pl-0 pr-20 md:text-left;
}

.main-info {
  @apply w-full mt-10 text-center;
  @apply md:mt-0 md:ml-5 md:text-left md:w-3/4;
  @apply lg:w-5/6;
}

.authors span::after {
  content: '-';
  position: relative;
  margin: 0 8px;
}

.authors span:last-of-type::after {
  display: none;
}
</style>
