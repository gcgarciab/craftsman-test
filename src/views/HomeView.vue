<template>
  <section class="home wrapper">
    <AppButton
      icon="heart"
      class="absolute top-5 right-5"
      title="Go to favorites"
      :type="ButtonType.BUTTON"
      :color="ButtonColor.DANGER"
      @click="$router.push({ name: 'Favorites' })"
    />

    <form @submit.prevent="searchBook()" class="books-search">
      <img
        src="../assets/images/craftsman-logo.png"
        alt="Craftman Logo"
        title="Craftman Logo"
        class="mb-10 w-3/4 lg:w-3/5"
      />

      <div class="form-row">
        <div class="search-input w-full">
          <AppInputField
            v-model="search"
            icon="search"
            placeholder="Search books by volume"
          />
        </div>

        <AppButton
          text="Search"
          :type="ButtonType.SUBMIT"
          :color="ButtonColor.PRIMARY"
          :disabled="allowSearchButton"
        />
      </div>
    </form>

    <div v-if="!mainStore.isLoading" class="books-result m-auto lg:w-5/6">
      <div
        v-if="bookStore.books.length > 0"
        class="book-cards flex flex-col m-auto"
      >
        <BookCard
          v-for="book in bookStore.books"
          :key="book.id"
          :id="book.id"
          :show-detail-link="true"
          :volume-info="book.volumeInfo"
          :is-favorite="favorites.includes(book.id)"
          @toggle-favorite="toggleFavoriteBook(book.id)"
        />

        <AppButton
          text="Load more"
          icon="spinner"
          :margin-top="true"
          :type="ButtonType.BUTTON"
          :color="ButtonColor.DEFAULT"
          @click="loadMoreBooks()"
        />
      </div>

      <ErrorMessage
        v-if="searchDone && bookStore.books.length === 0"
        title="Error!"
        description="Not found data, try again."
        icon="exclamation-triangle"
      />
    </div>

    <AppLoader v-else />
  </section>
</template>

<script setup lang="ts">
import { computed, ref } from 'vue';
import { storeToRefs } from 'pinia';

import useBookStore from '../stores/book';
import useMainStore from '../stores/main';
import ButtonType from '../enums/button-type';
import ButtonColor from '../enums/button-color';
import BookCard from '../components/BookCard.vue';
import ErrorMessage from '../components/ErrorMessage.vue';
import AppInputField from '../ui/AppInputField.vue';
import AppButton from '../ui/AppButton.vue';
import AppLoader from '../ui/AppLoader.vue';

const search = ref('');
const bookStore = useBookStore();
const mainStore = useMainStore();
const searchDone = ref(false);
const { favorites } = storeToRefs(bookStore);

const allowSearchButton = computed(
  () => search.value.length <= 2 || search.value === bookStore.currentSearch
);

/**
 * Calls fetchBooks action from store.
 * and allow isLoading component.
 */
async function searchBook(): Promise<void> {
  mainStore.$patch({ isLoading: true });

  await bookStore.fetchBooks(search.value).catch(() => mainStore.$reset());

  searchDone.value = true;
  mainStore.$reset();
}

/**
 * Calls fetchBooks action from book store specifying the page.
 */
function loadMoreBooks(): void {
  bookStore.fetchBooks(search.value, bookStore.currentPage + 1);
}

/**
 * Calls toggleFavoriteBook action from book store to add or
 * remove specific book from favorites state.
 * @param id => Book id
 */
function toggleFavoriteBook(id: string) {
  bookStore.toggleFavoriteBook(id);
}
</script>

<style scoped>
.books-search {
  @apply m-auto flex flex-col items-center;
  @apply lg:w-full;
}

.form-row {
  @apply flex flex-row items-start justify-between w-full;
  @apply lg:w-5/6;
}
</style>
