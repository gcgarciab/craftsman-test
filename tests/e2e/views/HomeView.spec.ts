import { mount } from '@vue/test-utils';
import { createTestingPinia, TestingOptions } from '@pinia/testing';
import HomeView from '@/views/HomeView.vue';
import useBookStore from '@/stores/book';
import AppButton from '@/ui/AppButton.vue';
import AppInputField from '@/ui/AppInputField.vue';
import data from '@/mocks/books-search.json';
import BookCard from '@/components/BookCard.vue';
import flushPromises from 'flush-promises';

jest.mock('axios', () => ({
  create: jest.fn(() => ({
    get: jest.fn(() => Promise.resolve({ data })),
    interceptors: {
      request: { use: jest.fn(), eject: jest.fn() },
      response: { use: jest.fn(), eject: jest.fn() },
    },
  })),
}));

describe('HomeView', () => {
  let wrapper: any;
  let store: any;

  function factory(options?: TestingOptions) {
    const wrapper = mount(HomeView, {
      global: {
        plugins: [createTestingPinia(options)],
        mocks: {
          $router: { push: jest.fn() }
        }
      },
    });

    const store = useBookStore();

    return { store , wrapper };
  }

  beforeEach(async () => {
    const testfactory = factory({ stubActions: false });
    store = testfactory.store;
    wrapper = testfactory.wrapper;

    const searchInputContainer = wrapper.find('.search-input')
      .getComponent(AppInputField).find('[data-test="input-field"]');
    
    await searchInputContainer.setValue('soccer');
    await wrapper.find('form').trigger('submit.prevent');
    await flushPromises();
  });

  afterEach(() => wrapper.unmount());

  it('Call fetchBooks action from store, and render results', async () => {
    expect(store.books.length).toBe(10);
    expect(wrapper.html()).toMatchSnapshot();
    expect(store.fetchBooks).toHaveBeenCalledTimes(1);
  });

  it('Call toggleFavoriteBook action on fetched book card', async () => {
    expect(store.favorites.length).toBe(0);

    const firstBookCard = wrapper.findAllComponents(BookCard).at(0);
    const favoriteButton = firstBookCard?.findAllComponents(AppButton).at(1);
    await favoriteButton?.trigger('click');
    await flushPromises();

    expect(store.favorites.includes('b-hH4Ma4Hq0C'));
    expect(store.favorites.length).toBe(1);
    
    await favoriteButton?.trigger('click');
    await flushPromises();

    expect(store.favorites.length).toBe(0);
  });

  it('Call fetchBooks action from store, using "load more" button', async () => {
    const loadMoreButton = wrapper.find('.books-result').find('[data-test="load-more"]');

    await loadMoreButton.trigger('click');
    await flushPromises();

    expect(store.books.length).toBe(20);
    expect(store.fetchBooks).toHaveBeenCalledTimes(2);
  });
});