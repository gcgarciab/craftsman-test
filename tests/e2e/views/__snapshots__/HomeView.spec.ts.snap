// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView Call fetchBooks action from store, and render results 1`] = `
"<section class=\\"home wrapper\\"><button class=\\"text-white px-2 flex items-center justify-center danger absolute top-5 right-5\\" data-test=\\"button\\" type=\\"button\\" title=\\"Go to favorites\\">
    <fa class=\\"icon text-sm\\" icon=\\"heart\\"></fa>
  </button>
  <form class=\\"books-search\\"><img alt=\\"Craftman Logo\\" title=\\"Craftman Logo\\" class=\\"mb-10 w-3/4 lg:w-3/5\\">
    <div class=\\"form-row\\">
      <div class=\\"search-input w-full\\">
        <div class=\\"input-block\\">
          <!--v-if-->
          <div class=\\"input\\">
            <fa class=\\"input-icon\\" icon=\\"search\\"></fa><input type=\\"text\\" placeholder=\\"Search books by volume\\" class=\\"with-icon\\" data-test=\\"input-field\\">
            <!--v-if-->
            <!--v-if-->
          </div>
        </div>
      </div><button class=\\"text-white px-10 inline primary\\" data-test=\\"button\\" type=\\"submit\\" disabled=\\"\\">
        <!--v-if--> Search
      </button>
    </div>
  </form>
  <div class=\\"books-result m-auto lg:w-5/6\\">
    <div class=\\"book-cards flex flex-col m-auto\\">
      <div class=\\"book-card\\">
        <div class=\\"book-header\\">
          <div class=\\"book-image w-36 h-48 flex justify-center items-center border\\"><img src=\\"http://books.google.com/books/content?id=b-hH4Ma4Hq0C&amp;printsec=frontcover&amp;img=1&amp;zoom=5&amp;edge=curl&amp;source=gbs_api\\" alt=\\"Book image\\" title=\\"Book image\\"></div>
          <div class=\\"main-info\\">
            <h3 class=\\"book-title\\">¡el Primer Libro de Fútbol para el Bebé!
              <!--v-if-->
            </h3>
            <h4 class=\\"authors\\"><span>Stephen Berg</span></h4>
            <p class=\\"published mt-3\\">Sole Books, 2012-11-14</p>
            <div class=\\"language mt-3\\"> Language: (ES)
              <!--v-if-->
            </div>
          </div>
        </div>
        <div class=\\"card-actions absolute right-6 top-6 w-20 flex justify-between\\"><button class=\\"text-white px-2 flex items-center justify-center primary\\" data-test=\\"book-detail\\" type=\\"button\\" title=\\"See more\\">
            <fa class=\\"icon text-sm\\" icon=\\"info-circle\\"></fa>
          </button><button class=\\"text-white px-2 flex items-center justify-center default\\" data-test=\\"add-to-favorites\\" type=\\"button\\" title=\\"Add to favorites\\">
            <fa class=\\"icon text-sm\\" icon=\\"heart\\"></fa>
          </button></div>
      </div>
      <div class=\\"book-card\\">
        <div class=\\"book-header\\">
          <div class=\\"book-image w-36 h-48 flex justify-center items-center border\\"><img src=\\"http://books.google.com/books/content?id=qtmxDwAAQBAJ&amp;printsec=frontcover&amp;img=1&amp;zoom=5&amp;edge=curl&amp;source=gbs_api\\" alt=\\"Book image\\" title=\\"Book image\\"></div>
          <div class=\\"main-info\\">
            <h3 class=\\"book-title\\">The Soccer Goalie <span class=\\"book-subtitle\\">El portero</span></h3>
            <h4 class=\\"authors\\"><span>Madison Capitano</span><span>Pablo de la Vega</span></h4>
            <p class=\\"published mt-3\\">Carson-Dellosa Publishing, 2020-01-07</p>
            <div class=\\"language mt-3\\"> Language: (ES) <span class=\\"total-pages mt-3\\"> - Pages: 32</span></div>
          </div>
        </div>
        <div class=\\"card-actions absolute right-6 top-6 w-20 flex justify-between\\"><button class=\\"text-white px-2 flex items-center justify-center primary\\" data-test=\\"book-detail\\" type=\\"button\\" title=\\"See more\\">
            <fa class=\\"icon text-sm\\" icon=\\"info-circle\\"></fa>
          </button><button class=\\"text-white px-2 flex items-center justify-center default\\" data-test=\\"add-to-favorites\\" type=\\"button\\" title=\\"Add to favorites\\">
            <fa class=\\"icon text-sm\\" icon=\\"heart\\"></fa>
          </button></div>
      </div>
      <div class=\\"book-card\\">
        <div class=\\"book-header\\">
          <div class=\\"book-image w-36 h-48 flex justify-center items-center border\\"><img src=\\"http://books.google.com/books/content?id=zt9fCQAAQBAJ&amp;printsec=frontcover&amp;img=1&amp;zoom=5&amp;edge=curl&amp;source=gbs_api\\" alt=\\"Book image\\" title=\\"Book image\\"></div>
          <div class=\\"main-info\\">
            <h3 class=\\"book-title\\">¡Cuenta conmigo! El torneo de fútbol (Count Me In! Soccer Tournament) 6-Pack
              <!--v-if-->
            </h3>
            <h4 class=\\"authors\\"><span>Lisa  Greathouse</span></h4>
            <p class=\\"published mt-3\\">Teacher Created Materials, 2012-03-30</p>
            <div class=\\"language mt-3\\"> Language: (ES) <span class=\\"total-pages mt-3\\"> - Pages: 28</span></div>
          </div>
        </div>
        <div class=\\"card-actions absolute right-6 top-6 w-20 flex justify-between\\"><button class=\\"text-white px-2 flex items-center justify-center primary\\" data-test=\\"book-detail\\" type=\\"button\\" title=\\"See more\\">
            <fa class=\\"icon text-sm\\" icon=\\"info-circle\\"></fa>
          </button><button class=\\"text-white px-2 flex items-center justify-center default\\" data-test=\\"add-to-favorites\\" type=\\"button\\" title=\\"Add to favorites\\">
            <fa class=\\"icon text-sm\\" icon=\\"heart\\"></fa>
          </button></div>
      </div>
      <div class=\\"book-card\\">
        <div class=\\"book-header\\">
          <div class=\\"book-image w-36 h-48 flex justify-center items-center border\\"><img src=\\"http://books.google.com/books/content?id=UOAiEAAAQBAJ&amp;printsec=frontcover&amp;img=1&amp;zoom=5&amp;edge=curl&amp;source=gbs_api\\" alt=\\"Book image\\" title=\\"Book image\\"></div>
          <div class=\\"main-info\\">
            <h3 class=\\"book-title\\">¡Hora de fútbol! (Soccer Time!)
              <!--v-if-->
            </h3>
            <h4 class=\\"authors\\"><span>Brendan Flynn</span></h4>
            <p class=\\"published mt-3\\">Lerner Digital ™, 2017-08-01</p>
            <div class=\\"language mt-3\\"> Language: (ES) <span class=\\"total-pages mt-3\\"> - Pages: 24</span></div>
          </div>
        </div>
        <div class=\\"card-actions absolute right-6 top-6 w-20 flex justify-between\\"><button class=\\"text-white px-2 flex items-center justify-center primary\\" data-test=\\"book-detail\\" type=\\"button\\" title=\\"See more\\">
            <fa class=\\"icon text-sm\\" icon=\\"info-circle\\"></fa>
          </button><button class=\\"text-white px-2 flex items-center justify-center default\\" data-test=\\"add-to-favorites\\" type=\\"button\\" title=\\"Add to favorites\\">
            <fa class=\\"icon text-sm\\" icon=\\"heart\\"></fa>
          </button></div>
      </div>
      <div class=\\"book-card\\">
        <div class=\\"book-header\\">
          <div class=\\"book-image w-36 h-48 flex justify-center items-center border\\"><img src=\\"http://books.google.com/books/content?id=3yfwAAAAMAAJ&amp;printsec=frontcover&amp;img=1&amp;zoom=5&amp;source=gbs_api\\" alt=\\"Book image\\" title=\\"Book image\\"></div>
          <div class=\\"main-info\\">
            <h3 class=\\"book-title\\">The Soccer Yellow Pages
              <!--v-if-->
            </h3>
            <h4 class=\\"authors\\"></h4>
            <!--v-if-->
            <div class=\\"language mt-3\\"> Language: (EN)
              <!--v-if-->
            </div>
          </div>
        </div>
        <div class=\\"card-actions absolute right-6 top-6 w-20 flex justify-between\\"><button class=\\"text-white px-2 flex items-center justify-center primary\\" data-test=\\"book-detail\\" type=\\"button\\" title=\\"See more\\">
            <fa class=\\"icon text-sm\\" icon=\\"info-circle\\"></fa>
          </button><button class=\\"text-white px-2 flex items-center justify-center default\\" data-test=\\"add-to-favorites\\" type=\\"button\\" title=\\"Add to favorites\\">
            <fa class=\\"icon text-sm\\" icon=\\"heart\\"></fa>
          </button></div>
      </div>
      <div class=\\"book-card\\">
        <div class=\\"book-header\\">
          <div class=\\"book-image w-36 h-48 flex justify-center items-center border\\"><img src=\\"http://books.google.com/books/content?id=SHMDEAAAQBAJ&amp;printsec=frontcover&amp;img=1&amp;zoom=5&amp;edge=curl&amp;source=gbs_api\\" alt=\\"Book image\\" title=\\"Book image\\"></div>
          <div class=\\"main-info\\">
            <h3 class=\\"book-title\\">¡Hora del fútbol! (Soccer Time! Spanish Edition)
              <!--v-if-->
            </h3>
            <h4 class=\\"authors\\"><span>Terry Pierce</span></h4>
            <p class=\\"published mt-3\\">Random House Books for Young Readers, 2021-07-06</p>
            <div class=\\"language mt-3\\"> Language: (ES) <span class=\\"total-pages mt-3\\"> - Pages: 32</span></div>
          </div>
        </div>
        <div class=\\"card-actions absolute right-6 top-6 w-20 flex justify-between\\"><button class=\\"text-white px-2 flex items-center justify-center primary\\" data-test=\\"book-detail\\" type=\\"button\\" title=\\"See more\\">
            <fa class=\\"icon text-sm\\" icon=\\"info-circle\\"></fa>
          </button><button class=\\"text-white px-2 flex items-center justify-center default\\" data-test=\\"add-to-favorites\\" type=\\"button\\" title=\\"Add to favorites\\">
            <fa class=\\"icon text-sm\\" icon=\\"heart\\"></fa>
          </button></div>
      </div>
      <div class=\\"book-card\\">
        <div class=\\"book-header\\">
          <div class=\\"book-image w-36 h-48 flex justify-center items-center border\\"><img src=\\"http://books.google.com/books/content?id=8uU_DwAAQBAJ&amp;printsec=frontcover&amp;img=1&amp;zoom=5&amp;edge=curl&amp;source=gbs_api\\" alt=\\"Book image\\" title=\\"Book image\\"></div>
          <div class=\\"main-info\\">
            <h3 class=\\"book-title\\">El futbol Soccer es divertido (Soccer is Fun)
              <!--v-if-->
            </h3>
            <h4 class=\\"authors\\"><span>Calee M. Lee</span></h4>
            <p class=\\"published mt-3\\">Xist Publishing, 2017-12-01</p>
            <div class=\\"language mt-3\\"> Language: (ES) <span class=\\"total-pages mt-3\\"> - Pages: 24</span></div>
          </div>
        </div>
        <div class=\\"card-actions absolute right-6 top-6 w-20 flex justify-between\\"><button class=\\"text-white px-2 flex items-center justify-center primary\\" data-test=\\"book-detail\\" type=\\"button\\" title=\\"See more\\">
            <fa class=\\"icon text-sm\\" icon=\\"info-circle\\"></fa>
          </button><button class=\\"text-white px-2 flex items-center justify-center default\\" data-test=\\"add-to-favorites\\" type=\\"button\\" title=\\"Add to favorites\\">
            <fa class=\\"icon text-sm\\" icon=\\"heart\\"></fa>
          </button></div>
      </div>
      <div class=\\"book-card\\">
        <div class=\\"book-header\\">
          <div class=\\"book-image w-36 h-48 flex justify-center items-center border\\"><img src=\\"http://books.google.com/books/content?id=TXXWww7KOFsC&amp;printsec=frontcover&amp;img=1&amp;zoom=5&amp;edge=curl&amp;source=gbs_api\\" alt=\\"Book image\\" title=\\"Book image\\"></div>
          <div class=\\"main-info\\">
            <h3 class=\\"book-title\\">La historia del fútbol (The Story of Soccer)
              <!--v-if-->
            </h3>
            <h4 class=\\"authors\\"><span>Anastasia Suen</span></h4>
            <p class=\\"published mt-3\\">The Rosen Publishing Group, Inc, 2003-12-15</p>
            <div class=\\"language mt-3\\"> Language: (ES) <span class=\\"total-pages mt-3\\"> - Pages: 24</span></div>
          </div>
        </div>
        <div class=\\"card-actions absolute right-6 top-6 w-20 flex justify-between\\"><button class=\\"text-white px-2 flex items-center justify-center primary\\" data-test=\\"book-detail\\" type=\\"button\\" title=\\"See more\\">
            <fa class=\\"icon text-sm\\" icon=\\"info-circle\\"></fa>
          </button><button class=\\"text-white px-2 flex items-center justify-center default\\" data-test=\\"add-to-favorites\\" type=\\"button\\" title=\\"Add to favorites\\">
            <fa class=\\"icon text-sm\\" icon=\\"heart\\"></fa>
          </button></div>
      </div>
      <div class=\\"book-card\\">
        <div class=\\"book-header\\">
          <div class=\\"book-image w-36 h-48 flex justify-center items-center border\\"><img src=\\"http://books.google.com/books/content?id=UEFgDwAAQBAJ&amp;printsec=frontcover&amp;img=1&amp;zoom=5&amp;edge=curl&amp;source=gbs_api\\" alt=\\"Book image\\" title=\\"Book image\\"></div>
          <div class=\\"main-info\\">
            <h3 class=\\"book-title\\">Es hora de jugar f�tbol (It's Time for the Soccer Game)
              <!--v-if-->
            </h3>
            <h4 class=\\"authors\\"><span>Sadie Woods</span></h4>
            <p class=\\"published mt-3\\">The Rosen Publishing Group, Inc, 2017-07-15</p>
            <div class=\\"language mt-3\\"> Language: (ES) <span class=\\"total-pages mt-3\\"> - Pages: 24</span></div>
          </div>
        </div>
        <div class=\\"card-actions absolute right-6 top-6 w-20 flex justify-between\\"><button class=\\"text-white px-2 flex items-center justify-center primary\\" data-test=\\"book-detail\\" type=\\"button\\" title=\\"See more\\">
            <fa class=\\"icon text-sm\\" icon=\\"info-circle\\"></fa>
          </button><button class=\\"text-white px-2 flex items-center justify-center default\\" data-test=\\"add-to-favorites\\" type=\\"button\\" title=\\"Add to favorites\\">
            <fa class=\\"icon text-sm\\" icon=\\"heart\\"></fa>
          </button></div>
      </div>
      <div class=\\"book-card\\">
        <div class=\\"book-header\\">
          <div class=\\"book-image w-36 h-48 flex justify-center items-center border\\"><img src=\\"http://books.google.com/books/content?id=j4BYAAAAYAAJ&amp;printsec=frontcover&amp;img=1&amp;zoom=5&amp;source=gbs_api\\" alt=\\"Book image\\" title=\\"Book image\\"></div>
          <div class=\\"main-info\\">
            <h3 class=\\"book-title\\">Soccer
              <!--v-if-->
            </h3>
            <h4 class=\\"authors\\"><span>V-Five Association of America</span></h4>
            <!--v-if-->
            <div class=\\"language mt-3\\"> Language: (EN) <span class=\\"total-pages mt-3\\"> - Pages: 192</span></div>
          </div>
        </div>
        <div class=\\"card-actions absolute right-6 top-6 w-20 flex justify-between\\"><button class=\\"text-white px-2 flex items-center justify-center primary\\" data-test=\\"book-detail\\" type=\\"button\\" title=\\"See more\\">
            <fa class=\\"icon text-sm\\" icon=\\"info-circle\\"></fa>
          </button><button class=\\"text-white px-2 flex items-center justify-center default\\" data-test=\\"add-to-favorites\\" type=\\"button\\" title=\\"Add to favorites\\">
            <fa class=\\"icon text-sm\\" icon=\\"heart\\"></fa>
          </button></div>
      </div><button class=\\"text-white mt-8 px-10 flex items-center justify-center default\\" data-test=\\"load-more\\" type=\\"button\\">
        <fa class=\\"icon text-sm mr-2\\" icon=\\"spinner\\"></fa> Load more
      </button>
    </div>
    <!--v-if-->
  </div>
</section>"
`;
